packages/shared/src/utils/getCollateralAssetIdFromHash.ts

```diff
@@ -0,0 +1,14 @@
+import { AssetHash, Hash256 } from '@explorer/types'
+
+import { CollateralAsset } from '../CollateralAsset'
+
+export function getCollateralAssetIdFromHash(
+  hash: AssetHash | Hash256 | string,
+  collateralAsset: CollateralAsset
+) {
+  if (hash.toString() !== collateralAsset.assetHash.toString()) {
```

Comment:
Consider using `AssetHash(hash.toString())` on the left side in case there is 0x missing etc.